### PR TITLE
bugfix: Fix IsVFNode check error in provisioner

### PR DESCRIPTION
**Pull request by mowangdk**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Fix IsVFNode check error in provisioner

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by mowangdk** [comment]
@huww98 ptal

**Review by huww98 — COMMENTED** [review]
Not ideal. But we can fix it later.
/lgtm
/hold
please unhold if you still want to proceed given the comments.

File: pkg/disk/cloud.go [hunk 1]
```diff
@@ -222,7 +222,7 @@ func attachDisk(ctx context.Context, tenantUserUID, diskID, nodeID string, isSha
 		devicePaths := calcNewDevices(before, after)
 
 		// BDF Disk Logical
-		if IsVFNode() && len(devicePaths) == 0 {
+		if !GlobalConfigVar.ControllerService && IsVFNode() && len(devicePaths) == 0 {
```
Comment (huww98) [inline pkg/disk/cloud.go:225]: After #993 , We can actually enable controller and node service in the same process. And using global variable is not good. Maybe we should introduce a new parameter to attachDisk(). We should replace all ADControllerEnable related branch with the new parameter.

**Pull request merged**